server/api-util/lineItemHelpers.js

```diff
@@ -0,0 +1,192 @@
+const Decimal = require('decimal.js');
+const has = require('lodash/has');
+const { types } = require('sharetribe-flex-sdk');
+const { Money } = types;
+
+const { convertMoneyToNumber, unitDivisor, convertUnitToSubUnit } = require('./currency');
+const { nightsBetween, daysBetween } = require('./dates');
+const LINE_ITEM_NIGHT = 'line-item/night';
+const LINE_ITEM_DAY = 'line-item/day';
+
+/** Helper functions for constructing line items*/
+
+/**
+ * Calculates lineTotal for lineItem based on quantity.
+ * The total will be `unitPrice * quantity`.
+ *
+ * @param {Money} unitPrice
+ * @param {int} quantity
+ *
+ * @returns {Money} lineTotal
+ */
+exports.calculateTotalPriceFromQuantity = (unitPrice, unitCount) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice).times(unitCount).toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+/**
+ * Calculates lineTotal for lineItem based on percentage.
+ * The total will be `unitPrice * (percentage / 100)`.
+ *
+ * @param {Money} unitPrice
+ * @param {int} percentage
+ *
+ * @returns {Money} lineTotal
+ */
+exports.calculateTotalPriceFromPercentage = (unitPrice, percentage) => {
```

Comment:
Can the percentage be anything? e.g. -200, 345?

Reply:
I think in theory it could be. On API side the limitations come from payin and payout sums which can't be negative and payin must be greater or equal to payout. If the percentage is calculated from just one lineItem (e.g. cleaning fee or something else small) even that big percentage might work inside those payin and payout limits.